server/bin/gen-certs

```diff
@@ -47,19 +81,153 @@ if [ ! -d $UPSTREAM_CERTS_HOME ]; then
     $SUDO mkdir -p $UPSTREAM_CERTS_HOME
 fi
 
-HOSTNAME=${HOSTNAME:-$(hostname)}
 
 if [ -f $CA_KEY ] && [ -f $CA_CERT ] && [ "$FORCECERT" != "1" ]; then
     echo "Certificates are already present."
 else
+    echo
+    echo "Creating CA private key password"
+    sudo su -c "echo $RANDOM > $CA_KEY_PASSWORD"
```

Comment:
Should these `echo`s be with a `-n`?  I know some programs can interpret the trailing newline as part of the actual password and thus making it impossible to type on the CLI.